util/bytes.go

```diff
@@ -49,3 +49,73 @@ func ByteContainsAny(b []byte, l [][]byte) bool {
 
 	return false
 }
+
+// BytesRoughlyContains returns true if all bytes from the given byte slice `input` exist in the
+// given `output` byte slice -- the elements must be found in order. This is basically the same as
+// what you can see in @lithammer's(1) fuzzysearch `Match` function (thank you to them!) but
+// converted to work for bytes and to not use a continuation block. Some examples:
+//
+// input 'aa', output 'b' = false
+// input 'aa', output 'bba' = false
+// input 'aa', output 'bbaa' = true
+// input 'aba', output 'bba' = false
+//
+// In the context of scrapligo this is basically used for "fuzzy" matching our inputs. This lets us
+// cope with devices that do things like the following srlinux banner entry output:
+//
+//		```
+//	 --{ !* candidate shared default }--[  ]--
+//	 A:srl# system banner login-banner "
+//	 ...my banner
+//	 ...has
+//	 ...some lines
+//	 ...that are neat
+//	 ..."
+//	 --{ !* candidate shared default }--[  ]--
+//
+// The "..." at the beginning of each line would historically be problematic for scrapli because in
+// a very brute force/ham-fisted way we would demand to read back exactly what we sent to the device
+// in the output -- so the "..." broke that. Not cool! This can be used to ensure that doesn't
+// happen!
+//
+// Note: @lithammer's fuzzy search `Match` function here:
+// https://github.com/lithammer/fuzzysearch/blob/master/fuzzy/fuzzy.go#L60-L83
+func BytesRoughlyContains(input, output []byte) bool {
+	switch diffLen := len(output) - len(input); {
+	case diffLen < 0:
+		// output is not long enough to hold all our inputs, so definitely not roughly contains!
+		return false
+	case diffLen == 0:
+		// diff is same length, can directly test equality
+		if bytes.Equal(input, output) {
+			return true
+		}
+	}
+
+	for _, inputChar := range input {
+		var shouldContinue bool
+
+		shouldContinue, output = innerBytesRoughlyContains(inputChar, output)
+
+		if shouldContinue {
+			continue
+		}
+
+		return false
+	}
+
+	return true
+}
+
+func innerBytesRoughlyContains(
```

Comment:
the func name sounds tricky to me. What is `inner` here? Looks like this is the output byte slice.
But then why is it inner?

Reply:
`outputBytesContainInputByte`?
It seems there is no roughliness here? we basically check if the output byte slice has an input byte and if it is the rest of the output becomes the new output.
At least this is how my brain parses the flow here, I might be off though

Reply:
ahah touche, défo no roughliness! was keeping the name similar to the "parent" func since I think/hope we wont use that anywhere else. and yeah, your understanding is correct!